src/utils/PackedStringLib.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+/// @notice Efficient library for encoding/decoding strings shorter than 32 bytes as one word.
+/// @notice Solidity has built-in functionality for storing strings shorter than 32 bytes in
+/// a single word, but it must determine at runtime whether to treat each string as one word
+/// or several. This introduces a significant amount of bytecode and runtime complexity to
+/// any contract storing strings.
+/// @notice When it is known in advance that a string will never be longer than 31 bytes,
+/// telling the compiler to always treat strings as such can greatly reduce extraneous runtime
+/// code that would have never been executed.
+/// @notice https://docs.soliditylang.org/en/v0.8.17/types.html#bytes-and-string-as-arrays
+/// @author Solmate (https://github.com/transmissions11/solmate/blob/main/src/utils/PackedStringLib.sol)
+library PackedStringLib {
+    error UnpackableString();
+
+    /// @dev Pack a 0-31 byte string into a bytes32.
+    /// @dev Will revert if string exceeds 31 bytes.
+    function packString(string memory unpackedString) internal pure returns (bytes32 packedString) {
+        uint256 length = bytes(unpackedString).length;
```

Comment:
Can't you do something like `bytes32(bytes(unpackedString))`?

The conversion would truncate the string. So you'd probably need the `if (length > 31)` check if truncation is not good enough.

Reply:
Also would be nice to define a custom type `type ShortString is bytes32;`